File: easy/283.move-zeroes.py
"""
[0,1,0,3,12]



zeros ids

[1,0,3,12,0]


[1,3,12,0,0] <--
"""


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        z_count = 0
        l = len(nums)-1
        i = 0
        while i <= l: 
            if nums[i] == 0:
                z_count += 1
            else:
                nums[i-z_count] = nums[i] #1-1=1,3-2=3,
            i += 1
        
        i = l - z_count + 1
        while i <= l:
                nums[i] = 0
                i += 1
            
            
        
        
        

File: easy/283.move-zeroes.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut i = 0;
        let mut j = 0;
        while i < nums.len() {
            if nums[i] != 0 {
                nums.swap(j, i);
                j += 1;
            }
            i += 1;
        }
    }
}